### PR TITLE
Provision fuse pods in a separate namespace from the user's namespace

### PR DESCRIPTION
**Pull request by JameelB**
## Description
Provision the fuse operator and deployment configs in a separate namespace from the user's namespace.

When the user provisions fuse from the service catalog, it will create a service instance in the user's namespace. The broker will then create a separate namespace, only accessible to an admin user, to deploy the operator and create the custom resource into.

## Progress
- [x] Create separate namespace for the fuse operator
- [x] Deploy Fuse operator into that namespace
- [x] Create Fuse custom resource into that namespace
- [x] Pass Fuse's dashboard url to the service instance created in the user's namespace
- [x] Allow the user who provisioned the service access to the Fuse dashboard

## Verification Steps

1. Create the Fuse [custom resource definition](https://github.com/syndesisio/syndesis/blob/master/install/operator/deploy/syndesis-crd.yml).
2. Follow the steps on how to [deploy the broker](https://github.com/aerogear/managed-services-broker/blob/provision-fuse/README.md#deploying-the-broker).
3. Verify that you can see the fuse plan in the service catalog.
4. Provision fuse.
5. Verify that a new namespace, only accessed by an admin user, has been created where the fuse operator and its deployment configs are created.
6. Verify that when provision completes, the dashboard url shows up on the service instance in the user namespace.
7. Verify that the user can access the dashboard url.

### CODE REVIEW

File: pkg/deploys/fuse/deployer.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+	glog.Infof("Getting fuse catalog entries")
+	return getCatalogServicesObj()
+}
+
+func (fd *FuseDeployer) GetID() string {
+	return fd.id
+}
+
+func (fd *FuseDeployer) Deploy(instanceID string, contextProfile brokerapi.ContextProfile, k8sclient kubernetes.Interface, osClientFactory *openshift.ClientFactory) (*brokerapi.CreateServiceInstanceResponse, error) {
+	glog.Infof("Deploying fuse from deployer, id: %s", instanceID)
```
Comment (david-martin) [inline pkg/deploys/fuse/deployer.go]: It might be useful to log out any errors coming from the blocks below

**Comment by david-martin** [comment]
I'm seeing an error when i provision fuse from the catalog

image

```
The service failed. Error provisioning ServiceInstance of ClusterServiceClass (K8S: "fuse-service-id" ExternalName: "fuse") at ClusterServiceBroker "managed-services-broker": Status: 400; ErrorMessage: <nil>; Description: <nil>; ResponseError: <nil>
```

After some digging (thanks @JameelB) it looks like the error is actually the code in `createFuseCustomResource` where it expects a particular route to exist in the `default` project, and that route doesn't exist in my local cluster.

@maleck13 I wonder if there's a better way to try figure out the route the fuse operator will ultimately create so it can be returned as the dashboard url.
My understanding is we can't wait for the route to be created by the operator before sending back the dashboard url as it would take too long, 
and the dashboard url can't be set later after the first request to the broker (i.e. it can't be set later by the last operation stuff when the provision eventually finishes)

**Comment by david-martin** [comment]
> I wonder if there's a better way to try figure out the route the fuse operator will ultimately create so it can be returned as the dashboard url.

@JameelB Thinking on this more, I believe the managed services broker should be pre-configured with the `ROUTING_SUFFIX` for the cluster it's installed in. Setting it should be a 1 time thing at installation of the managed services broker. Then the broker only has to read this config when it needs to figure out what a dashboard url for a new fuse instance will be e.g. `https://fuse-<NAMESPACE>.<ROUTING_SUFFIX>`, rather than looking up another route and parsing out the suffix from that. 

How does this sound?

**Comment by david-martin** [comment]
@JameelB could we break out the vendoring and the custom changes in different PRs?
That would make reviewing easier.

Also, a suggestion from @maleck13, is all of syndesis needed to be vendored?
Is it enough to just vendor a specific folder that has the types?

**Comment by JameelB** [comment]
@david-martin sounds good to me. I can add the routing suffix as a parameter when creating the template so it can be registered as an environment variable. I'll create a separate PR for the vendor changes. 

We don't really need the rest of syndesis apart from the types but I'm not sure if we can vendor just the specific folder? When i import the types and ran `dep ensure` it vendored out all of syndesis automatically. Is there a way of just including the types? @maleck13

File: README.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ oc new-project managed-services-broker
 oc create -f deploy/rbac.yaml
 
 # Process the template and create the broker deployment
-oc process -f templates/broker.template.yaml <-p IMAGE_ORG=yourDockerOrg> | oc create -f -
+oc process -f templates/broker.template.yaml -p ROUTE_SUFFIX=<clusterRouteSubDomain> <-p IMAGE_ORG=yourDockerOrg> | oc create -f -
```
Comment (david-martin) [inline README.md]: `<-p IMAGE_ORG=yourDockerOrg>` => `-p IMAGE_ORG=<yourDockerOrg>`

**Comment by david-martin** [comment]
@JameelB I'll give this a spin again today

**Review by david-martin — APPROVED** [review]
Verified in a local cluster.

After provision of fuse from the catalog, a new project is created as expected
```
oc get route syndesis -n fuse-d3170867-c04d-4974-84b0-102974bcb9d0
NAME       HOST/PORT                                                       PATH      SERVICES              PORT      TERMINATION          WILDCARD
syndesis   fuse-d3170867-c04d-4974-84b0-102974bcb9d0.192.168.42.1.nip.io             syndesis-oauthproxy   8443      reencrypt/Redirect   None
```

The ServiceInstance has the correct dashboardURL
```
svcat get instances -n myproject fuse-nrhr6 -o yaml | grep dashboardURL
  dashboardURL: https://fuse-d3170867-c04d-4974-84b0-102974bcb9d0.192.168.42.1.nip.io
```

I can login to the fuse dashboard OK.

**Pull request merged**